scripts/kconfig/kconfigfunctions.py

```diff
@@ -415,6 +415,27 @@ def dt_node_str_prop_equals(kconf, _, path, prop, val):
 
     return "n"
 
+def dt_node_array_prop_len(kconf, _, label, prop):
+    """
+    This function takes a 'label' and looks for an EDT node for that label. If
+    it finds an EDT node, it will look to see if that node has a property
+    by the name of 'prop'.  If the 'prop' exists and is an array type it will
+    return the length of its array as a string integer value, or 0 if the
+    property does not exist.
+    """
+    if doc_mode or edt is None:
+        return "0"
+
+    try:
+        node = edt.label2node.get(label)
+    except edtlib.EDTError:
+        return "0"
+
+    if node and (prop in node.props) and (node.props[prop].type == "array"):
```

Comment:
Do you maybe want to consider other array types too? phandle-array, uint8-array, etc?

Maybe `"array" in node.props[prop].type` instead, in other words.